modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/consensus/ConsensusStateUpdater.scala

```diff
@@ -65,359 +60,212 @@ object ConsensusStateUpdater {
 
   def make[F[
     _
-  ]: Async: KryoSerializer: Hasher: SecurityProvider: Metrics, Event, Key: Show: Order: TypeTag: Encoder, Artifact <: AnyRef: Eq: TypeTag: Encoder, Context <: AnyRef: Eq: TypeTag](
-    consensusFns: ConsensusFunctions[F, Event, Key, Artifact, Context],
-    consensusStorage: ConsensusStorage[F, Event, Key, Artifact, Context],
+  ]: Async: KryoSerializer: Hasher: SecurityProvider: Metrics, Event, Key: Show: Order: TypeTag: Encoder, Artifact <: AnyRef: Eq: TypeTag: Encoder, Context <: AnyRef: Eq: TypeTag, Status: Eq: Show, Outcome: Eq, Kind: Encoder: Eq: Show: TypeTag](
+    consensusStateAdvancer: ConsensusStateAdvancer[F, Key, Artifact, Context, Status, Outcome, Kind],
+    consensusStorage: ConsensusStorage[F, Event, Key, Artifact, Context, Status, Outcome, Kind],
     gossip: Gossip[F],
-    keyPair: KeyPair
```

Comment:
It isn't clear here but what happened here is mostly pulling out the status handling logic to a separate class (so that we get custom consensus flow). A lot of functionalities are marked as red and then with green (moved rather than changed)